regparser/grammar/utils.py

```diff
@@ -117,6 +125,9 @@ def initial_regex(grammar):
         if (isinstance(grammar, pyparsing.And) and
                 isinstance(grammar.exprs[0], pyparsing.Optional)):
             return recurse(grammar.exprs[0].expr) | recurse(grammar.exprs[1])
+        if (isinstance(grammar, pyparsing.And) and
+                isinstance(grammar.exprs[0], pyparsing.Empty)):
+            return recurse(grammar.exprs[1])
```

Comment:
Indeed. The new solution for `keep_pos` is to use
```
Empty + OriginalGrammar + Empty
```
attaching position information to the wrapping `Empty`s. We hadn't seen `Empty` (which reads zero characters) before, so the "Quick" version didn't exist.